test/inductor/mock_cache.py

```diff
@@ -1,127 +1,114 @@
-        self._clients = {}  # Used for Manifold
+if TYPE_CHECKING:
+    from types import TracebackType
+
 
-    def reset(self):
-        self.num_init = 0
+@dataclasses.dataclass
+class Stats:
+    num_put: int = 0
+    num_get_hit: int = 0
+    num_get_miss: int = 0
+
+    def __iadd__(self, other: Stats) -> Self:
+        self.num_put += other.num_put
+        self.num_get_hit += other.num_get_hit
+        self.num_get_miss += other.num_get_miss
+        return self
+
+    def reset(self) -> None:
         self.num_put = 0
         self.num_get_hit = 0
         self.num_get_miss = 0
 
-    def report(self):
-        print(
-            "".join(
-                [
-                    f"{self._name} cache: ",
-                    f"init: {self.num_init}, ",
-                    f"puts: {self.num_put}, ",
-                    f"misses: {self.num_get_miss}, ",
-                    f"hits: {self.num_get_hit}, ",
-                ]
-            ),
-            file=sys.stderr,
+    def __str__(self) -> str:
+        return "".join(
+            (
+                f"puts: {self.num_put}, ",
+                f"misses: {self.num_get_miss}, ",
+                f"hits: {self.num_get_hit}, ",
+            )
         )
 
 
-class _MockLocalAutotuneCacheBackend(RemoteCacheBackend):
-    _state = _MockCacheState("Local")
+# The cache states are thread-local so if we're running multiple tests at once
+# they won't cross contaminate. However - it needs to be "global" because we
+# allow code to create new cache clients which refer to the same cache (because
+# it's a remote cache).
 
-    def __init__(self):
-        state = self._state
-        state.num_init += 1
 
-    @override
-    def get(self, key: str) -> Optional[bytes]:
-        assert isinstance(key, str)
+class _GlobalStats(Stats, threading.local):
+    def __init__(self) -> None:
+        self.autotune = Stats()
+        self.fx_graph = Stats()
+        self.triton = Stats()
 
-        state = self._state
-        if key in state._cache:
-            state.num_get_hit += 1
-            return state._cache[key]
-        else:
-            state.num_get_miss += 1
+    def reset(self) -> None:
+        self.autotune.reset()
+        self.fx_graph.reset()
+        self.triton.reset()
 
-    @override
-    def put(self, key: str, data: bytes) -> None:
-        assert isinstance(key, str)
-        assert isinstance(data, bytes)
+    def update(self, name: str, delta: Stats) -> None:
+        stat = getattr(self, name)
+        stat += delta
 
-        state = self._state
-        state.num_put += 1
-        state._cache[key] = data
+    def report(self):
+        print("Cache Stats:", file=sys.stderr)
+        print(f"  autotune: {self.autotune}", file=sys.stderr)
+        print(f"  fx_graph: {self.fx_graph}", file=sys.stderr)
+        print(f"  triton:   {self.triton}", file=sys.stderr)
 
 
-class _MockRedisRemoteCache:
-    _state = _MockCacheState("Redis")
+global_stats = _GlobalStats()
```

Comment:
This is not indented correctly.

Reply:
What do you mean, this is intended to be top level

Reply:
See https://github.com/pytorch/pytorch/actions/runs/10602712425/job/29386272600?pr=134032#step:20:645

Reply:
it was on line 108. But now we have new push...

Reply:
You are correct. I threw in some debugging prints to try to figure out why a test is failing and messed up the indent on them. Unfortunately it's an internal PR which sometimes takes some time to push changes out.